api/src/main/java/com/spotify/ffwd/util/HighFrequencyDetector.java

```diff
@@ -0,0 +1,157 @@
+/*-
+ * -\-\-
+ * FastForward API
+ * --
+ * Copyright (C) 2020 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.ffwd.util;
+
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import com.spotify.ffwd.model.Metric;
+import com.spotify.ffwd.statistics.HighFrequencyDetectorStatistics;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class HighFrequencyDetector {
+
+  /** Allow to drop high frequency metrics. */
+  @Inject
+  @Named("dropHighFrequencyMetric")
+  Boolean dropHighFrequencyMetric;
```

Comment:
is it legit for these fields to be nullabe? just wondering if the primitive types could be used instead

Reply:
Good point, I've changed it to use primitive as I default in case it's not set.